libpod/container_log.go

```diff
@@ -93,17 +95,20 @@ func (c *Container) readFromLogFile(ctx context.Context, options *logs.LogOption
 	}()
 	// Check if container is still running or paused
 	if options.Follow {
+		// If the container isn't running or if we encountered an error
+		// getting its state, instruct the logger to read the file
+		// until EOF.
 		state, err := c.State()
 		if err != nil || state != define.ContainerStateRunning {
-			// If the container isn't running or if we encountered
-			// an error getting its state, instruct the logger to
-			// read the file until EOF.
+			// Make sure to wait at least for the poll duration
+			// before stopping the file logger (see #10675).
+			time.Sleep(watch.POLL_DURATION)
```

Comment:
I'm always worried about putting a hard sleep into a function like this, but I'm not thinking of a better way atm.

Reply:
Same here. I _really_ did not like it which is why I removed it previously.  I read the docs of the library we're using but they did not reflect reality :(